### DISCUSSION TITLE
Orion EdgeConnect SWS08HA Outdoor Siren

### DISCUSSION THREAD

**Discussion by ghost** [make-all/tuya-local#681] [Q&A]
Hello All

I'm fairly new to Home Assistant.

I've recently added two of these sirens to Home Assistant using official Tuya and Tuya-Local.

https://www.bunnings.com.au/orion-grid-connect-smart-outdoor-siren_p0261692

Strangely the official Tuya integration lets me control the siren's settings (siren volume, battery level etc.) but does not actually activate the siren, and it does not show itself as a siren device).

However, adding the devices in Tuya-Local exposes the actual siren and allows the siren to be activated and used in Alarmo.

So between Tuya Offical and Tuya-Local I can get these to work.   I'm not sure if there is a better way to use these which doesn't rely on both Tuya Official and Tuya-Local.

However, that is a side issue, and I'm having another issue adding a third one of these sirens.

While adding these devices in Tuya-Local, the first two of these sirens gave me the option during the Tuya-Local configuration to select between SWS08HA Outdoor Siren and a loratap_curtain_switch.

I just chose the Siren option, and adding the first two sirens worked as described above.

However, adding a THIRD siren, the only option now displayed during configuration is the loratap_curtain_switch.

![image](https://user-images.githubusercontent.com/125883873/236667946-8f974f97-4be6-419d-ad71-1fbec0dcc6c6.png)

So obviously using that option does not work.

Perhaps there's some simple issue I'm overlooking, but otherwise perhaps there's a bug when adding three of the same device?  Perhaps there's a way to manually configure these devices to force it to detect a SWS08HA Outdoor Siren?

As I say, I am fairly new to this so I'm not sure what other information might shed more light on this.

Any suggestions to fix this?

As another side note, upgrading from HA 2023.4.6 to 2023.5.1 seemed to completely break the integration of this siren, so I've actually downgraded back to 2023.4.6.   Another side issue, but perhaps someone might be able to shed some light on this as well.

Thank you!

**Comment by make-all** [comment]
You shouldn't need the cloud Tuya to change the settings, as the siren entity will let you control volume, tone and duration, and the config for this siren has a battery level sensor, as well as binary sensors for charging and tamper.

For the siren that cannot be detected correctly, the log message would be helpful, as that will show what values the siren is sending out, which will help diagnose why it is not correctly detected.

I haven't observed any issues with HA 2023.5.x and this integration, though there have been issues with other integrations due to long deprecated functions being removed.  Again, if you have any log messages from the upgrade it would help to see what is broken, as I do not have every type of device supported by this integration, and there may be issues in areas of the code that are unexercised by my install.

**Comment by ghost** [comment]
Thanks for responding so quickly.

Looking at the logs, my Unifi Video cameras seemed to be generating quite a few warnings so I removed them and restarted HA in order to clean up the logs a bit.

However, after restarting the third Siren was added to Tuya-Local in the same manner as the previous two.  Perhaps restarting HA fixed something.

Here is a snippet from the log which seems to relate to this not working before the restart, which might reveal what was going wrong:


2023-05-07 17:53:37.590 ERROR (MainThread) [homeassistant.components.cover] Error while setting up tuya_local platform for cover
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 304, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/tuya_local/cover.py", line 21, in async_setup_entry
    await async_tuya_setup_platform(
  File "/config/custom_components/tuya_local/helpers/config.py", line 28, in async_tuya_setup_platform
    data[ecfg.config_id] = entity_class(device, ecfg)
  File "/config/custom_components/tuya_local/cover.py", line 53, in __init__
    if "stop" in self._control_dp.values(self._device):
TypeError: argument of type 'NoneType' is not iterable
2023-05-07 17:53:48.494 WARNING (MainThread) [homeassistant.components.camera] Updating uvc camera took longer than the scheduled update interval 0:00:30
2023-05-07 17:53:54.122 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry Siren - Tool Shed for cover
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 383, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/usr/src/homeassistant/homeassistant/components/cover/__init__.py", line 207, in async_setup_entry
    return await component.async_setup_entry(entry)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 169, in async_setup_entry
    raise ValueError("Config entry has already been setup!")
ValueError: Config entry has already been setup!
2023-05-07 17:54:08.346 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry Siren - Tool Shed for cover
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 383, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/usr/src/homeassistant/homeassistant/components/cover/__init__.py", line 207, in async_setup_entry
    return await component.async_setup_entry(entry)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 169, in async_setup_entry
    raise ValueError("Config entry has already been setup!")
ValueError: Config entry has already been setup!
2023-05-07 17:54:08.347 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry Siren - Tool Shed for cover
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 383, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/usr/src/homeassistant/homeassistant/components/cover/__init__.py", line 207, in async_setup_entry
    return await component.async_setup_entry(entry)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 169, in async_setup_entry
    raise ValueError("Config entry has already been setup!")
ValueError: Config entry has already been setup!



While this issue seems to have resolved, I am still only able to set the Siren's time and volume via the official Tuya integration, as follows:

![image](https://user-images.githubusercontent.com/125883873/236675778-a67004a7-2a8f-4bee-98b4-efca88e2f632.png)

Whereas from within the Tuya-Local integration I get access to 4 entities, being the Siren control, as well as diagnostic information battery, charging and tamper detect as follows:

![image](https://user-images.githubusercontent.com/125883873/236675822-9a3ff3a7-a666-4ac7-8555-6430cccf8238.png)

Could this be to do with the order in which I added these to HA?  The device (and all other Tuya Official devices) are added automatically to the Tuya Official integration once I add them to the Tuya Smart app and only show the options set out above.

Then I add them to Tuya-Local in order to access the Siren control.

Would removing Tuya-Official help perhaps?

I would remove Tuya-Official, but I need it as its the only way to integrate the door and window sensors I currently have.   These are same brand as the Siren (any chance of these getting support in Tuya-Local? or know if these should be accessible without Tuya Official? https://www.bunnings.com.au/orion-grid-connect-smart-security-contact-sensor_p0261689)

Again, sorry I'm fairly new and perhaps I'm doing something wrong or just don't know what I'm doing!

As for what happened updating to 2023.5.1 - I'll need to look into this a bit more and see what I can find.

**Comment by make-all** [comment]
Here is the documentation for the siren entity. https://www.home-assistant.io/integrations/siren/
The volume and duration can be set as arguments to the siren.turn_on service call.  If you want them to be controls in the UI, I would suggest making some Helpers, and a Button that calls siren.turn_on using the volume and duration taken from those helper controls.

**Reply by ghost** [reply]
Thanks will have a look at this.

Seems like a good next step in learning about HA.  

Cheers
